test/routes_test.go

```diff
@@ -390,17 +390,26 @@ func TestRouteQueueSemantics(t *testing.T) {
 	routeSend("MSG foo RSID:1:1 2\r\nok\r\n")
 	// Queue group one.
 	routeSend("MSG foo QRSID:1:2 2\r\nok\r\n")
+	// Invlaid queue sid.
+	routeSend("MSG foo QRSID 2\r\nok\r\n")
+	routeSend("MSG foo QRSID:1 2\r\nok\r\n")
+	routeSend("MSG foo QRSID:1: 2\r\nok\r\n")
 
 	// Use ping roundtrip to make sure its processed.
 	routeSend("PING\r\n")
 	routeExpect(pongRe)
 
-	// Should be 2 now, 1 for all normal, and one for specific queue subscriber.
-	matches = clientExpectMsgs(2)
+	// Should be 3 now, 1 for all normal, and one for specific queue subscriber,
```

Comment:
As explained above, that should not change the number of messages delivered to regular subscribers, so you would need to revert those changes.